espnet2/asr/frontend/adapters_utils/adapters/adapter.py

```diff
@@ -0,0 +1,45 @@
+import torch
+import torch.nn as nn
+from fairseq import utils
+from fairseq.modules import LayerNorm
+
+
+class Adapter(nn.Module):
+    def __init__(
+        self,
+        orig_dim: int,
+        down_dim: int,
+        layer_norm: str = None,
```

Comment:
Did applying layer_norm help? In the `adapter_transformer`, this layer_norm is not used. Can you put some information about this argument?

Reply:
yes, the `layer_norm` in `adapter.py` is different from `layer_norm` in `adapter_transformer.py`:
- `layer_norm` in `adapter_transformer.py` is the layer norm in normal transformer layer.
-  `layer_norm` in `adapter` themselves is a bit complicated: There currently exist two implementations for `adapter` module themselves: 
 1.  The original [Adapter paper](https://arxiv.org/pdf/1902.00751.pdf) does not use the `layer_norm` **inside** adapters
 2. Whereas the [speech adapter paper](https://arxiv.org/pdf/2202.03218.pdf) (figure 2) uses `layer_norm` **inside** adapters.

Do you think we should do 1 or 2?

Reply:
Did you compare the performance of those? If not, I think we can do 1 as default.